app/frontend/src/features/messages/groupchats/GroupChatView.tsx

```diff
@@ -1,94 +1,164 @@
     {
       onSuccess: () => {
-        queryClient.invalidateQueries([
-          "groupChatMessages",
-          groupChat.groupChatId,
-        ]);
+        queryClient.invalidateQueries(["groupChatMessages", groupChatId]);
         queryClient.invalidateQueries(["groupChats"]);
       },
     }
   );
 
   const leaveGroupChatMutation = useMutation<Empty, GrpcError, void>(
-    () => service.conversations.leaveGroupChat(groupChat.groupChatId),
+    () => service.conversations.leaveGroupChat(groupChatId),
     {
       onSuccess: () => {
-        queryClient.invalidateQueries([
-          "groupChatMessages",
-          groupChat.groupChatId,
-        ]);
+        queryClient.invalidateQueries(["groupChatMessages", groupChatId]);
         queryClient.invalidateQueries(["groupChats"]);
       },
     }
   );
 
+  const history = useHistory();
+
   const handleLeaveGroupChat = () => leaveGroupChatMutation.mutate();
+  const handleBack = () => history.goBack();
   return (
     <Box className={classes.root}>
-      <Typography variant="h3">{groupChat!.title}</Typography>
-      <Button onClick={closeGroupChat}>
-        <CloseIcon />
-        (close)
-      </Button>
-      <Button onClick={handleLeaveGroupChat}>
-        <CloseIcon />
-        (leave)
-      </Button>
-      {(error || sendMutation.error || leaveGroupChatMutation.error) && (
-        <Alert severity="error">
-          {error ||
-            sendMutation.error?.message ||
-            leaveGroupChatMutation.error?.message}
-        </Alert>
-      )}
-      {isLoading ? (
-        <CircularProgress />
+      {!groupChatId ? (
+        <Alert severity="error">Invalid chat id.</Alert>
       ) : (
-        messages && (
-          <>
-            <MessageList messages={messages} />
-            <SendField sendMutation={sendMutation} />
-          </>
-        )
+        <>
+          <Box className={classes.header}>
+            <HeaderButton onClick={handleBack} aria-label="Back">
+              <BackIcon />
+            </HeaderButton>
+
+            <PageTitle className={classes.title}>
+              {groupChat ? (
+                groupChatTitleText(
+                  groupChat,
+                  groupChatMembersQuery,
+                  currentUserId
+                )
+              ) : idError ? (
+                "Error"
+              ) : (
+                <Skeleton width={100} />
+              )}
+            </PageTitle>
+
+            <HeaderButton
+              onClick={handleClick}
+              aria-label="Menu"
+              aria-haspopup="true"
+              innerRef={menuAnchor}
+            >
+              <SettingsIcon />
+            </HeaderButton>
+            <Menu
+              id="simple-menu"
+              anchorEl={menuAnchor.current}
+              keepMounted
+              open={menuOpen}
+              onClose={handleClose}
+            >
+              <MenuItem onClick={handleLeaveGroupChat}>Leave chat</MenuItem>
```

Comment:
The sides of this popup on this is right against the text, so could do with some side padding?

Reply:
Done, this is because of the override I put in theme.ts... 🤔